generators/entity-client/templates/angular/src/main/webapp/app/entities/entity.service.ts.ejs

```diff
@@ -45,91 +45,86 @@ export class <%= entityAngularName %>Service {
-            .map((res: EntityArrayResponseType) => this.convertArrayResponse(res));
+            <%_ if (hasDate) { _%>.map((res: EntityArrayResponseType) => this.convertDateArrayFromServer(res))<%_ } _%>;
     }
     <%_ } _%>
 
-    private convertResponse(res: EntityResponseType): EntityResponseType {
-        const body: I<%= entityAngularName %> = this.convertItemFromServer(res.body);
-        return res.clone({body});
-    }
-
-    private convertArrayResponse(res: EntityArrayResponseType): EntityArrayResponseType {
-        const jsonResponse: I<%= entityAngularName %>[] = res.body;
-        const body: I<%= entityAngularName %>[] = [];
-        for (let i = 0; i < jsonResponse.length; i++) {
-            body.push(this.convertItemFromServer(jsonResponse[i]));
-        }
-        return res.clone({body});
-    }
-
-    /**
-     * Convert a returned JSON object to <%= entityAngularName %>.
-     */
-    private convertItemFromServer(<%= entityInstance %>: I<%= entityAngularName %>): I<%= entityAngularName %> {
+    <%_ if (hasDate) { _%>
+    private convertDateFromClient(<%= entityInstance %>: I<%= entityAngularName %>): I<%= entityAngularName %> {
         const copy: I<%= entityAngularName %> = Object.assign({}, <%= entityInstance %>, {
         <%_ for (idx in fields) { _%>
         <%_ if ( ['Instant', 'ZonedDateTime', 'LocalDate'].includes(fields[idx].fieldType) ) { _%>
-        <%=fields[idx].fieldName%>: <%= entityInstance %>.<%= fields[idx].fieldName %> != null ? moment(<%= entityInstance %>.<%= fields[idx].fieldName %>) : <%= entityInstance %>.<%= fields[idx].fieldName %>,
-            <%_ } _%>
-        <%_ } _%>
+        <%= fields[idx].fieldName %>: <%= entityInstance %>.<%= fields[idx].fieldName %> != null && <%= entityInstance %>.<%= fields[idx].fieldName %>.isValid() ? <%= entityInstance %>.<%= fields[idx].fieldName %>.toJSON() : null,
         });
+        <%_ } _%>
+        <%_ } _%>
         return copy;
     }
+    <%_ } _%>
 
-    /**
-     * Convert a <%= entityAngularName %> to a JSON which can be sent to the server.
-     */
-    private convert(<%= entityInstance %>: I<%= entityAngularName %>): I<%= entityAngularName %> {
-        const copy: I<%= entityAngularName %> = Object.assign({}, <%= entityInstance %>, {
-        <%_ for (idx in fields){ if ( ['Instant', 'ZonedDateTime', 'LocalDate'].includes(fields[idx].fieldType) ) { _%>
-        <%= fields[idx].fieldName %>: <%= entityInstance %>.<%= fields[idx].fieldName %> != null && <%= entityInstance %>.<%= fields[idx].fieldName %>.isValid() ? <%= entityInstance %>.<%= fields[idx].fieldName %>.toJSON() : null,
-        <%_ } } _%>
-        });
-        return copy;
+    <%_ if (hasDate) { _%>
+    private convertDateFromServer(res: EntityResponseType): EntityResponseType {
+        res.body.date = res.body.date != null ? moment(res.body.date) : res.body.date;
+        return res;
     }
+    <%_ } _%>
+
+    <%_ if (hasDate) { _%>
```

Comment:
merge with previous condition, as all methods are needed only when hasDate is true